src/main/java/com/asem/regestration/model/dao/impl/CourseDAOImpl.java

```diff
@@ -0,0 +1,86 @@
+package com.asem.regestration.model.dao.impl;
+
+import com.asem.regestration.model.dao.CourseDAO;
+import com.asem.regestration.model.dao.utils.DataSourcePool;
+import com.asem.regestration.model.entity.Courses;
+import com.asem.regestration.model.entity.Student;
+
+import java.sql.*;
+
+public class CourseDAOImpl implements CourseDAO {
+    private String userName;
+    private String passwored;
+    private String host;
+    private String databaseName;
+    private DataSourcePool dataSourcePool;
+
+
+    public CourseDAOImpl(String userName, String passwored, String host, String databaseName) {
+        this.userName=userName;
+        this.passwored=passwored;
+        this.host=host;
+        this.databaseName=databaseName;
+        this.dataSourcePool=new DataSourcePool(userName,passwored,host,databaseName);
+
+    }
+    public Connection getConnection() throws SQLException {
+
+        return dataSourcePool.getConnections().getConnection();
+
+    }
+
+
+
+
+
+    @Override
+    public Courses findById(int id) throws SQLException {
+        String sql="select *from student2.course where course.courseId="+id+"";
+
+        PreparedStatement preparedStatement=getConnection().prepareStatement(sql);
+        ResultSet resultSet=preparedStatement.executeQuery();
+        resultSet.next();
+        int iD=resultSet.getInt(1);
+        String name=resultSet.getString(2);
+        String teacherName=resultSet.getString(3);
+        Courses courses=new Courses(iD,name,teacherName);
+
+
+
```

Comment:
no need to add noisy empty lines between method one empty line after each method is better